### PR TITLE
[CommandLine][NFCI] Do not add 'All' to 'RegisteredSubCommands'

### PR DESCRIPTION
**Pull request by igorkudrin**
After #75679, it is no longer necessary to add the `All` pseudo subcommand to the list of registered subcommands. The change causes the list to contain only real subcommands, i.e. an unnamed top-level subcommand and named ones. This simplifies the code a bit by removing some checks for this special case.

### CODE REVIEW

**Review by nhaehnle — APPROVED** [review]

**Review by jh7370 — COMMENTED** [review]
Not sure I fully follow this, but happy with others approving.

File: llvm/lib/Support/CommandLine.cpp [hunk 1]
```diff
@@ -348,15 +345,14 @@ class CommandLineParser {
...
-    if (sub != &SubCommand::getAll()) {
-      for (auto &E : SubCommand::getAll().OptionsMap) {
-        Option *O = E.second;
-        if ((O->isPositional() || O->isSink() || O->isConsumeAfter()) ||
-            O->hasArgStr())
-          addOption(O, sub);
-        else
-          addLiteralOption(*O, sub, E.first());
-      }
+    assert(sub != &SubCommand::getAll());
```
Comment (RKSimon) [inline llvm/lib/Support/CommandLine.cpp:348]: (style) assertion message

File: llvm/lib/Support/CommandLine.cpp [hunk 2]
```diff
@@ -532,8 +527,7 @@ SubCommand *CommandLineParser::LookupSubCommand(StringRef Name,
   // Find a subcommand with the edit distance == 1.
   SubCommand *NearestMatch = nullptr;
   for (auto *S : RegisteredSubCommands) {
-    if (S == &SubCommand::getAll())
-      continue;
+    assert(S != &SubCommand::getAll());
```
Comment (RKSimon) [inline llvm/lib/Support/CommandLine.cpp:530]: (style) assertion message

**Comment by igorkudrin** [comment]
* Add assertion messages

**Review by MaskRay — APPROVED** [review]

**Pull request merged**